rust/src/io/exec/knn.rs

```diff
@@ -120,8 +132,8 @@ impl KNNFlatExec {
     /// Create a new [KNNFlatExec] node.
     ///
     /// Returns an error if the preconditions are not met.
-    pub fn try_new(input: Arc<dyn ExecutionPlan>, query: Query) -> Result<Self> {
-        let schema = input.schema();
+    pub fn try_new(inputs: Vec<Arc<dyn ExecutionPlan>>, query: Query) -> Result<Self> {
```

Comment:
What is the case where `KNNFlatExec` has multiple children execute plan? Should it be  its child to take care the multiple sources are?

Reply:
you'd need to add a concat node? feels like unnecessary complexity for now? what use case are you anticipating with multiple child nodes

Reply:
oh  my question was that i dont see a case that KNNFlatExec has multiple inputs, so wanted to know why change the signature from `input` to `inputs: Vec<_>`. 

Because the number of children is kind important for the execution plan contract.  It feels wrong that KNNFlatExec needs to handle multiple children.